File: rest-classic/src/test/java/com/example/PostRepositoryTest.java
package com.example;

import com.example.repository.PostRepository;
import io.quarkus.test.junit.QuarkusTest;
import io.quarkus.test.junit.TestProfile;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.logging.Level;
import java.util.logging.Logger;

import static org.junit.jupiter.api.Assertions.assertEquals;

@QuarkusTest
//@QuarkusTestResource(H2DatabaseTestResource.class)
@TestProfile(H2Profile.class)
public class PostRepositoryTest {

    private final static Logger LOGGER = Logger.getLogger(PostRepositoryTest.class.getName());

    @Inject
    private PostRepository posts;

    @Test
    public void testPersistence() {
        this.posts.listAll().forEach(p -> LOGGER.log(Level.INFO, "Post:{0}", p));
        assertEquals(2, this.posts.findAll().list().size(), "result list size is 2");
    }

}

File: mongodb-kotlin-co/src/test/kotlin/com/example/PostRepositoryTest.kt
package com.example.entity

import com.example.repository.Post
import com.example.PostRepository
import io.kotest.matchers.shouldBe
import io.kotest.matchers.shouldNotBe
import io.quarkus.test.junit.QuarkusTest
import io.smallrye.mutiny.coroutines.awaitSuspending
import jakarta.inject.Inject
import kotlinx.coroutines.test.runTest
import org.junit.jupiter.api.Test
import java.util.logging.Level
import java.util.logging.Logger

@QuarkusTest
class PostRepositoryTest {
    companion object {
        private val log = Logger.getLogger(PostRepositoryTest::class.java.name)
    }

    @Inject
    lateinit var postRepository: PostRepository

    @Test
    fun `test save and query`() = runTest {
        val entity = Post(title = "foo", body = "bar")
        val saved = postRepository.persist(entity).awaitSuspending()
        saved.id shouldNotBe null

        val post = postRepository.findByTitle("foo").awaitSuspending()
        log.log(Level.INFO, "found post: $post")

        post shouldNotBe null
        post!!.title shouldBe "foo"
    }
}